mlflow/server/js/src/common/utils/AuthService.js

```diff
@@ -0,0 +1,99 @@
+export class AuthService {
+
+  token = null;
+  toRedirectState = '';
+  ssoLoginHeader = 'X-SSO-Login';
+  ssoLogoutHeader = 'X-SSO-Logout';
+  redirectUriRoute = 'oauth';
+
+  constructor() {
+    const savedToken = localStorage.getItem('token');
+    if (savedToken !== null) {
+      this.token = savedToken;
+    }
+  }
+
+  logOff() {
+    this.setNewTokenInternal(null);
+    const fullLogoutUrl = this.getFullLogoutUrl();
+    if (fullLogoutUrl !== null) {
+      window.location.assign(fullLogoutUrl);
+    } else {
+      window.location.reload();
+    }
+  }
+
+  getFullLogoutUrl() {
+    const logoutUrl = localStorage.getItem('logoutUrl');
+    if (logoutUrl !== null) {
+      return logoutUrl
+        + '&goto='
+        + encodeURIComponent(window.location.protocol + '//' + window.location.host + '/');
+    }
+    return null;
+  }
+
+  tryFetchTokenFromQueryParam(qpName) {
+    const urlParams = new URLSearchParams(document.location.search);
+    if (urlParams.has('token')) {
+      this.setNewToken(urlParams.get('token'));
+      urlParams.delete('token');
+      const otherParams = urlParams.toString();
+      otherParams !== ''
+        ? window.history.replaceState('', '', `?${urlParams.toString()}`)
+        : window.history.replaceState('', '', '');
+    }
+  }
```

Comment:
tryFetchTokenFromQueryParam cound be tested too... there are some logic that deserves a unit test

Reply:
Removed it because the function was never used